docs/architecture/lb/nginx.md

```diff
@@ -0,0 +1,270 @@
+# Test message with IETF frame for port 601/TCP:
+echo "11 hello world" | netcat <LB_IP> 601
+```
+
+3. Run performance tests based on the [Check TCP Performance](performance-tests.md#check-your-tcp-performance) section.
+
+| Receiver                   | Performance        |
+|----------------------------|--------------------|
+| Single SC4S Server         | 71,738.98 msg/sec  |
+| Load Balancer + 2 Servers  | 99,089.03 msg/sec  |
+
+Please note that load balancer fine-tuning is beyond the scope of the SC4S team's responsibility. For assistance in increasing the TCP throughput of your load balancer instance, contact the NGINX support team.
+
+## Option 2: Configure NGINX with DSR (Direct Server Return)
+
+### Advantages:
+- Works for UDP
+- Reduced latency
+
+### Disadvantages:
+- DSR setup requires active health checks because the load balancer cannot expect responses from the upstream. Active health checks are not available in NGINX, so switch to NGINX Plus or implement your own active health checking.
+- Requires switching to the `root` user.
+- For cloud users, this might require disabling `Source/Destination Checking` (tested with AWS).
+
+1. In the main NGINX configuration, update the `user` to root:
+`/etc/nginx/nginx.conf`
+```conf
+user root;
+```
+
+2. Add a configuration similar to the following in:
+
+**For NGINX Open Source:**
+
+`/etc/nginx/modules-enabled/sc4s.conf`
+```conf
+stream {
+    # Define upstream for each of SC4S hosts and ports
+    # Default SC4S UDP port is 514
+    # Include your custom ports if applicable
+    upstream stream_syslog_514 {
+        server <SC4S_IP_1>:514;
+        server <SC4S_IP_2>:514;
+    }
+
+    # Define connections to each of your upstreams.
+    # Ensure to include `proxy_bind` and `proxy_responses 0`.
+    server {
+        listen        514 udp;
+        proxy_pass    stream_syslog_514;
+        
+        proxy_bind $remote_addr:$remote_port transparent;
+        proxy_responses 0;
+    }
+}
+```
+
+**For NGINX Plus:**
+
+1. Add the following configuration block to `/etc/nginx/nginx.conf`:
+```conf
+stream {
+    # Define upstream for each of SC4S hosts and ports
+    # Default SC4S UDP port is 514
+    # Include your custom ports if applicable
+    upstream stream_syslog_514 {
+        zone   stream_syslog_514 64k;
+        server <SC4S_IP_1>:514;
+        server <SC4S_IP_2>:514;
+    }
+
+    # Define connections to each of your upstreams.
+    # Ensure to include `proxy_bind` and `health_check`.
+    server {
+        listen        514 udp;
+        proxy_pass    stream_syslog_514;
+        
+        proxy_bind $remote_addr:$remote_port transparent;
+
+        health_check udp;
+    }
+}
+```
+
+NGINX will actively check the health of your upstream servers by sending UDP messages to port 514.
+
+2. (Optional) Add the following local post-filter to each of your SC4S instances to prevent SC4S from forwarding health check messages to Splunk and other destinations.
```

Comment:
2. (Optional) Add the following local post-filter to each of your SC4S instances to prevent SC4S from forwarding health check messages to Splunk and other destinations: